openedx/core/djangoapps/schedules/content_highlights.py

```diff
@@ -0,0 +1,86 @@
+from courseware.module_render import get_module_for_descriptor
+from courseware.model_data import FieldDataCache
+from openedx.core.djangoapps.schedules.config import COURSE_UPDATE_WAFFLE_FLAG
+from openedx.core.djangoapps.schedules.exceptions import CourseUpdateDoesNotExist
+from request_cache import get_request_or_stub
+
+from xmodule.modulestore.django import modulestore
+
+
+def course_has_highlights(course_key):
+    """
+    Does the course have any highlights for any section/week in it?
+    This ignores access checks, since highlights may be lurking in currently
+    inaccessible content.
+    """
+    if not COURSE_UPDATE_WAFFLE_FLAG.is_enabled(course_key):
```

Comment:
Why does the waffle flag checking need to occur in `course_has_highlights` and `get_week_highlights`? 

It seems like we shouldn't allow the use of a feature's API if the feature is turned off. What do you think?

If so, maybe these would be better as assertions, because they are preconditions that must be met to use the function.

Reply:
discussed out of band.  decided to keep the waffle checks in both external functions.